osu.Game.Rulesets.Taiko/Difficulty/TaikoLegacyScoreSimulator.cs

```diff
@@ -109,21 +103,24 @@ private void simulateHit(HitObject hitObject)
                 case Swell swell:
                     // The taiko swell generally does not match the osu-stable implementation in any way.
                     // We'll redo the calculations to match osu-stable here...
-                    double minimumRotationsPerSecond = IBeatmapDifficultyInfo.DifficultyRange(playableBeatmap.Difficulty.OverallDifficulty, 3, 5, 7.5);
-                    double secondsDuration = swell.Duration / 1000;
 
-                    // The amount of half spins that are required to successfully complete the spinner (i.e. get a 300).
-                    int halfSpinsRequiredForCompletion = (int)(secondsDuration * minimumRotationsPerSecond);
+                    // Normally, this value depends on the final overall difficulty. For simplicity, we'll only consider the worst case that maximises rotations.
+                    const double minimum_rotations_per_second = 7.5;
 
+                    // The amount of half spins that are required to successfully complete the spinner (i.e. get a 300).
+                    int halfSpinsRequiredForCompletion = (int)(swell.Duration / 1000 * minimum_rotations_per_second);
                     halfSpinsRequiredForCompletion = (int)Math.Max(1, halfSpinsRequiredForCompletion * 1.65f);
 
-                    if (mods.Any(m => m is ModDoubleTime))
-                        halfSpinsRequiredForCompletion = Math.Max(1, (int)(halfSpinsRequiredForCompletion * 0.75f));
-                    if (mods.Any(m => m is ModHalfTime))
-                        halfSpinsRequiredForCompletion = Math.Max(1, (int)(halfSpinsRequiredForCompletion * 1.5f));
+                    //
+                    // Normally, this multiplier depends on the active mods (DT = 0.75, HT = 1.5). For simplicity, we'll only consider the worst case that maximises rotations.
+                    // This way, scores remain beatable at the cost of the conversion being slightly inaccurate.
+                    //   - A perfect DT/NM score will have less than 1M total score (excluding bonus).
+                    //   - A perfect HT score will have 1M total score (excluding bonus).
+                    //
+                    halfSpinsRequiredForCompletion = Math.Max(1, (int)(halfSpinsRequiredForCompletion * 1.5f));
```

Comment:
Hmm, on a second check, aren't changes similar to https://github.com/ppy/osu/pull/24779/commits/ed295effc5206fcb347d2e73cadc5c5abb9251db required here too?

Reply:
Unless I'm missing something, I don't believe so because swell ticks are base score not bonus?

Reply:
Hmm ok, that they are. My bad.